lbry/lbry/extras/daemon/Components.py

```diff
@@ -136,86 +146,60 @@ def component(self):
-                    if not chunk:
-                        break
-                    fd.write(chunk)
-                    self._headers_progress_percent = math.ceil(
-                        float(fd.tell()) / float(final_size_after_download) * 100
-                    )
-            log.info("fetched headers from s3, now verifying integrity after download.")
-            self._check_header_file_integrity()
-
-    def local_header_file_height(self):
-        return max((self.local_header_file_size() / HEADER_SIZE) - 1, 0)
+            buffer, header_size = b'', self.headers.header_size
+            async for chunk in response.content.iter_any():
+                chunk = buffer + chunk
+                remaining = len(chunk) % header_size
+                chunk, buffer = chunk[:-remaining], bytes(chunk[-remaining:])
+                if not chunk:
+                    continue
+                if not await self.headers.connect(len(self.headers), chunk):
+                    log.warning("Error connecting downloaded headers from at %s.", self.headers.height)
+                    return
+                self._headers_progress_percent = math.ceil(
+                    float(self.headers.bytes_size) / float(final_size_after_download) * 100
+                )
 
     def local_header_file_size(self):
         if os.path.isfile(self.headers_file):
             return os.stat(self.headers_file).st_size
         return 0
 
-    async def get_remote_height(self):
-        ledger = SimpleNamespace()
-        ledger.config = {
-            'default_servers': self.conf.lbryum_servers,
-            'data_path': self.conf.wallet_dir
-        }
-        net = Network(ledger)
-        first_connection = net.on_connected.first
-        asyncio.ensure_future(net.start())  # TODO: SKETCHY! it might be trapping a CancelledError and not raising it
-        await first_connection
-        remote_height = await net.get_server_height()
-        await net.stop()
-        return remote_height
+    async def get_download_height(self):
+        async with utils.aiohttp_request('HEAD', HEADERS_URL) as response:
+            if response.status != 200:
+                log.warning("Header download error: %s", response.status)
+                return 0
+            return response.content_length // HEADER_SIZE
 
     async def should_download_headers_from_s3(self):
         if self.conf.blockchain_name != "lbrycrd_main":
             return False
-        self._check_header_file_integrity()
         s3_headers_depth = self.conf.s3_headers_depth
         if not s3_headers_depth:
             return False
-        local_height = self.local_header_file_height()
-        remote_height = await self.get_remote_height()
+
+        local_height = self.headers.height
+        try:
+            remote_height = await self.get_download_height()
+        except OSError:
+            log.warning("Failed to download headers using https.")
+            return False
```

Comment:
If you change `get_download_height()` (renamed to `get_downloadable_header_height()`) to return `None` when a height couldn't be fetched then this could just be:
```
remote_height = await self.get_downloadable_header_height()
if remote_height is None:
    log.warning("Failed to download headers using https.")
    return False
```